builder/vsphere/supervisor/step_watch_source.go

```diff
@@ -165,7 +171,7 @@ func (s *StepWatchSource) waitForVMReady(ctx context.Context, logger *PackerLogg
 				logger.Info("Source VM is NOT powered-on yet, continue watching...")
 			}
 
-		case <-time.After(time.Duration(s.Config.WatchSourceTimeoutSec) * time.Second):
+		case <-s.timeoutCh:
```

Comment:
To ensure each step gets the full elapsed time. 
```suggestion
		case <-time.After(time.Duration(s.Config.WatchSourceTimeoutSec) * time.Second):
```

Reply:
Resolving this comment as it's intended.